chapters/en-UK/xctest/types_of_testing.md

```diff
@@ -1,21 +1,23 @@
 # Types of Testing
 
-A test is a way of verifying code, and making your assumptions explicit. When you make a change somewhere, tests exist to show the implicit connections between objects.
+A test is a way of verifying code, and making your assumptions explicit. Tests exist to show connections between objects. When you make a change somewhere, tests reveal implicit connections.
 
-In the ruby world the tests are basically considered the application's documentation. This is one of the first port of calls for understanding how systems work after reading the overview/README.
+In the Ruby world the tests are basically considered the application's documentation. This is one of the first port of calls for understanding how systems work after reading the overview/README. It's quite similar with node. Both languages are interpreted languages where even variable name typos only show themselves in runtime errors. Without a compiler, you can catch this stuff only with rigorous testing.
 
-In Cocoa, this is less of a problem as we can rely on the tooling more. Given a compiler, static type systems, and a well integrated static analyser - you can know if your code is going to break someone else's very quickly. However, this is only a lint, the code compiling doesn't mean that it will still correctly show a view controller when you tap a button.
+In Cocoa, this is less of a problem as we can rely on the tooling more. Given a compiler, static type systems, and a well integrated static analyser -- you can know if your code is going to break someone else's very quickly. However, this is only a lint, the code compiling doesn't mean that it will still correctly show a view controller when you tap a button. In other words, the behaviour of your app can still be wrong.
 
-That is what testing is for. I will cover three major topics in this book - there are many, many other types of testing patterns ( for example, last week I was introduced to Model Testing. ) However, These are the dominent patterns that exist within the Cocoa world, and I have experience with them. Otherwise, I'd just be copy & pasting from Wikipedia.
+That is what testing is for. I will cover three major topics in this book. There are many, many other types of testing patterns -- for example, last week I was introduced to Model Testing. However, these three topics are the dominent patterns that exist within the Cocoa world, and I have experience with them. Otherwise, I'd just be copy & pasting from Wikipedia.
```

Comment:
I think that's BDD->xUnit a lossy transition though, whereas xUnit->BDD isn't. The main thing that I care about here is the `describe` and  `before`/`after` structure that you get that makes it significantly easier to write smaller, more focused tests. 

You can obviously refactor to make all your own functions to work around that but the xUnit flatness makes mean that your code is now totally intermingled with helper functions and `testXXX` functions.